api/rosetta/balance.go

```diff
@@ -48,50 +46,26 @@ func (d *Data) Balance(ctx echo.Context) error {
 	var req BalanceRequest
 	err := ctx.Bind(&req)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, invalidEncoding(invalidJSON, err))
+		return unpackError(err)
 	}
 
 	err = d.validate.Request(req)
 	if err != nil {
-		return validationError(err)
+		return formatError(err)
 	}
 
 	// TODO: Check if we can set up validation middleware to remove the
 	// redundant business logic between routes:
 	// => https://github.com/optakt/flow-dps/issues/164
 
 	err = d.config.Check(req.NetworkID)
```

Comment:
This should probably be moved into the validator, with the config being injected into the validator to be able to check the network ID appropriately.

Reply:
I was thinking the same thing 👍